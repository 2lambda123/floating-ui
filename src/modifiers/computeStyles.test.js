// @flow
import {
  mapStrategyToPosition,
  computePopperStyles,
  computeArrowStyles,
} from './computeStyles';

it('returns the expected position value', () => {
  expect(mapStrategyToPosition('fixed')).toBe('fixed');
  expect(mapStrategyToPosition('absolute')).toBe('absolute');
});

it('computes the popper styles', () => {
  window.devicePixelRatio = 2;
  expect(
    computePopperStyles({
      offsets: { x: 10, y: 5 },
      strategy: 'absolute',
      gpuAcceleration: true,
    })
  ).toMatchSnapshot();

  expect(
    computePopperStyles({
      offsets: { x: 10, y: 5 },
      strategy: 'absolute',
      gpuAcceleration: false,
    })
  ).toMatchSnapshot();
  window.devicePixelRatio = 1;
});

it('computes the arrow styles', () => {
  expect(
    computeArrowStyles({
      offsets: { x: 10, y: 5 },
      gpuAcceleration: true,
    })
  ).toMatchSnapshot();
});
