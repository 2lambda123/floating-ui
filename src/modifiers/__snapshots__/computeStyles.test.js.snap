// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`computes the arrow styles 1`] = `
Object {
  "bottom": "",
  "left": "0",
  "position": "absolute",
  "right": "",
  "top": "0",
  "transform": "translate(10px, 5px)",
}
`;

exports[`computes the popper styles 1`] = `
Object {
  "bottom": "",
  "left": "0",
  "position": "absolute",
  "right": "",
  "top": "0",
  "transform": "translate3d(10px, 5px, 0)",
}
`;

exports[`computes the popper styles 2`] = `
Object {
  "bottom": "",
  "left": "10px",
  "position": "absolute",
  "right": "",
  "top": "5px",
  "transform": "",
}
`;

exports[`computes the popper styles 3`] = `
Object {
  "bottom": "105px",
  "left": "10px",
  "position": "absolute",
  "right": "",
  "top": "",
  "transform": "",
}
`;

exports[`computes the popper styles 4`] = `
Object {
  "bottom": "",
  "left": "",
  "position": "absolute",
  "right": "110px",
  "top": "5px",
  "transform": "",
}
`;
